operator/cluster/cluster.go

```diff
@@ -202,6 +190,56 @@ func AddClusterBase(clientset *kubernetes.Clientset, client *rest.RESTClient, cl
 
 }
 
+//getOrCreatePVC will first verify whether the pgcluster name matches the "current primary" CRD value of the pgcluster CRD.
+//Based on this information, the underlying PVC name for the primary pod can be validated. If the expected PVC exists,
+//the correct PVC name is returned.
+//If the "current primary" label and the pgcluster name do match, this is either the recreation of a cluster that has not
+//previously failed over, or it is a new pgcluster. If a recreation, the PVC is reused, otherwise a new PVC is allocated.
+func getOrCreatePVC(clientset *kubernetes.Clientset, cl *crv1.Pgcluster, namespace string) string {
```

Comment:
But this doesn't appear to be "getting" a PVC, it's returning the name of the primary.  This also seems to change the meaning of the prior code.

I think if you're returning a name, you should be returning a *pvcName* which is what it seems the prior code expects.

Additionally, we shouldn't be publishing a "cluster failure" from here. That should be part of bubbling up errors to the primary function, and having that handle said errors.

Reply:
The function has been renamed to better reflect what it does. 
The PVC name no longer needs to be returned because of how the 'current primary' name is being stored. 
The errors are now passed upward to be handled in the AddClusterBase function.